clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -717,7 +716,7 @@ private byte[] serializeComputeRequest(ComputeRequestWrapper computeRequestWrapp
   public void start() throws VeniceClientException {
     if (needSchemaReader) {
       this.schemaReader = new RouterBackedSchemaReader(
-          this::getStoreClientForSchemaReader,
+          this,
```

Comment:
Essentially, try the following combination:
1. Sending `get` call to the backend.
2. Backend returns a value with an unknown schema id.
3. `callback` in the get future will try to retrieve the schema via schema reader.
In the past, that won't work for httpasyncclient (we should have a test for this).
Can you check it again?

Reply:
I added a test case and it seems to work. I might have misunderstood your suggestion, though. Can you take a look at my test and verify if it's what you wanted me to test? https://gist.github.com/nisargthakkar/1168d4516e55c8740f0e6dd4433c1e0a (I added this to `TestHybrid` class)

Reply:
Discussed and investigated offline. Here are my findings:

I found that internally, there are multiple workers, each with their own queue to process certain events like connection establishment, session creation, etc. When the worker finds some element in the queue, the event is processed in the same thread that polls the queue. This includes execution of the user-level callback function.

I observed that all the requests I created to the same route would always get added to the same worker's queue. This leads to a deadlock now because:
1. The callback function is waiting for the new request to complete
2. The new request is waiting in the queue to be picked up for processing
3. The thread that polls the queue is waiting for the callback to complete

This could be a bug in httpclients impl and we can start a conversation with them, but right now, it is not an issue for us because we are using `handleAsync` to perform further tasks on the future's completion. This will ensure that the handling is done on a different thread than the original executor thread